src/components/datepicker/datePicker.spec.js

```diff
@@ -25,29 +35,28 @@ describe('md-date-picker', function() {
     pageScope.myDate = initialDate;
     pageScope.isDisabled = false;
 
-    var template = '<md-datepicker ' +
-          'md-max-date="maxDate" ' +
-          'md-min-date="minDate" ' +
-          'ng-model="myDate" ' +
-          'ng-required="isRequired" ' +
-          'ng-disabled="isDisabled">' +
-      '</md-datepicker>';
+    createDatepickerInstance(DATEPICKER_TEMPLATE);
+    controller.closeCalendarPane();
+  }));
+
+  /** Compile and link the given template and store values for element, scope, and controller. */
+  function createDatepickerInstance(template) {
     ngElement = $compile(template)(pageScope);
```

Comment:
Optional: Can we instead always set ngElement to the datepicker element and return this element from this function instead if the caller needs to use it? I think it might be a little misleading to have ngElement potentially refer to something other than the datepicker element.